lib/twofishes/geocode_request.rb

```diff
@@ -0,0 +1,57 @@
+module Twofishes
+  class GeocodeRequest < ::GeocodeRequest
+    def initialize(options = {})
+      options[:ll] = prepare_ll(options[:ll])
+      options[:bounds] = prepare_bounds(options[:bounds])
+
+      options = substitute_aliases(options)
+
+      super(options)
+    end
+
+    def prepare_ll(ll)
+      case ll
+      when String
+        lat, lng = ll.split(/\s*,\s*/)
+        GeocodePoint.new(lat: lat.to_f, lng: lng.to_f)
+      when Array
+        GeocodePoint.new(lat: ll[0].to_f, lng: ll[1].to_f)
+      when Hash
+        GeocodePoint.new(lat: (ll[:lat] || ll['lat']).to_f, lng: (ll[:lng] || ll['lng'] || ll[:lon] || ll['lon']).to_f)
+      else
+        ll
+      end
+    end
+
+    def prepare_bounds(bounds)
+      case bounds
+      when String
+        ne_lat, ne_lng, sw_lat, sw_lng = bounds.split(/\s*,\s*/)
+        ne = GeocodePoint.new(lat: ne_lat.to_f, lng: ne_lng.to_f)
+        sw = GeocodePoint.new(lat: sw_lat.to_f, lng: sw_lng.to_f)
+        GeocodeBoundingBox.new(ne: ne, sw: sw)
+      when Array
+        ne = GeocodePoint.new(lat: bounds[0].to_f, lng: bounds[1].to_f)
+        sw = GeocodePoint.new(lat: bounds[2].to_f, lng: bounds[3].to_f)
+        GeocodeBoundingBox.new(ne: ne, sw: sw)
+      when Hash
+        ne = GeocodePoint.new(lat: (bounds[:ne_lat] || bounds['ne_lat']).to_f, lng: (bounds[:ne_lng] || bounds['ne_lng'] || bounds[:ne_lon] || bounds['ne_lon']).to_f)
+        sw = GeocodePoint.new(lat: (bounds[:sw_lat] || bounds['sw_lat']).to_f, lng: (bounds[:sw_lng] || bounds['sw_lng'] || bounds[:sw_lon] || bounds['sw_lon']).to_f)
+        GeocodeBoundingBox.new(ne: ne, sw: sw)
+      else
+        bounds
+      end
+    end
+
+    def substitute_aliases(options)
+      options[:maxInterpretations] = options.delete(:max) || options.delete(:max_interpretations)
```

Comment:
Shall we assign with `||=` in order to make sure the most exact option does not get overwritten? Eg. if I would have `{maxInterpretations: 5, max: 999}` it would take 5 and not overwrite it by 999?